adapter/internal/api/apis_process.go

```diff
@@ -0,0 +1,212 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"strings"
+
+	"github.com/wso2/product-microgateway/adapter/config"
+	"github.com/wso2/product-microgateway/adapter/internal/loggers"
+	"github.com/wso2/product-microgateway/adapter/internal/oasparser/constants"
+	"github.com/wso2/product-microgateway/adapter/internal/oasparser/model"
+	"github.com/wso2/product-microgateway/adapter/internal/oasparser/utills"
+	"github.com/wso2/product-microgateway/adapter/pkg/tlsutils"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	openAPIDir                 string = "Definitions"
+	openAPIFilename            string = "swagger."
+	apiYAMLFile                string = "api.yaml"
+	deploymentsYAMLFile        string = "deployment_environments.yaml"
+	endpointCertFile           string = "endpoint_certificates."
+	apiJSONFile                string = "api.json"
+	endpointCertDir            string = "Endpoint-certificates"
+	interceptorCertDir         string = "Endpoint-certificates/interceptors"
+	crtExtension               string = ".crt"
+	pemExtension               string = ".pem"
+	apiTypeFilterKey           string = "type"
+	apiTypeYamlKey             string = "type"
+	lifeCycleStatus            string = "lifeCycleStatus"
+	securityScheme             string = "securityScheme"
+	endpointImplementationType string = "endpointImplementationType"
+	inlineEndpointType         string = "INLINE"
+	endpointSecurity           string = "endpoint_security"
+	production                 string = "production"
+	sandbox                    string = "sandbox"
+	zipExt                     string = ".zip"
+)
+
+// ValidateAPIType checks if the apiProject is properly assigned with the type.
+func ValidateAPIType(apiProject *model.ProjectAPI) (err error) {
```

Comment:
rename project_structs.go as project.go. then move `parseDeployments`, `ValidateAPIType` and `ProcessFileInsideProject` there

Reply:
Thanks for pointing out. Currently the method `ProcessFileInsideProject` are only used in `api_impl.go`. Therefore, let me rename it to a private method. That way both `processFileInsideProject` and `parseDeployments` can stay inside the package `api`. As for the method `ValidateAPIType`, this can be also used in `server.go` once we start reading asyncAPI files. Since the API type validation happens based on apiyaml, updating the input param to `apiYaml` and moving the method to `apiYaml.go`